examples/flutter_gallery/lib/welcome/home.dart

```diff
@@ -0,0 +1,153 @@
+    ];
+    _tabController =
+        TabController(initialIndex: 0, length: _steps.length, vsync: this);
+    _pageController = PageController();
+    _autoProgressTimer = _scheduleAutoProgressStepTimer();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final MediaQueryData mediaQueryData = MediaQuery.of(context);
+    final double footerHeight = footerButtonHeight +
+        (Platform.isIOS ? mediaQueryData.padding.bottom : 0.0);
+    return Material(
+      color: _kWelcomeBlue,
+      child: NotificationListener<ScrollNotification>(
+        onNotification: (ScrollNotification notification) {
+          _autoProgressTimer.cancel();
+          _autoProgressTimer = _scheduleAutoProgressStepTimer();
+        },
+        child: Stack(
+          children: <Widget>[
+            Positioned.fill(
+              child: Padding(
+                padding: EdgeInsets.only(bottom: footerHeight),
+                child: PageView.builder(
+                  itemCount: _steps.length,
+                  itemBuilder: (BuildContext context, int index) =>
+                      _steps[index].contentWidget(),
+                  controller: _pageController,
+                  onPageChanged: (int page) {
+                    _tabController.animateTo(page);
+                    _steps[page].animate(restart: true);
+                  },
+                ),
+              ),
+            ),
+            Align(
+              alignment: FractionalOffset.bottomCenter,
+              child: Padding(
+                padding: EdgeInsets.only(bottom: footerHeight + 20.0),
+                child: _pageIndicator(),
+              ),
+            ),
+            Positioned.fill(
+              top: null,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: <Widget>[
+                  Container(
+                    height: footerButtonHeight,
+                    child: FlatButton(
+                      shape: const BeveledRectangleBorder(
+                        borderRadius: BorderRadius.only(
+                          topLeft: Radius.circular(16.0),
+                          topRight: Radius.circular(16.0),
+                        ),
+                      ),
+                      color: footerColor,
+                      child: const Text(
+                        'START EXPLORING',
+                        style: footerButtonTextStyle,
+                      ),
+                      onPressed: () {
+                        _autoProgressTimer.cancel();
+                        if (widget.onDismissed != null) {
+                          widget.onDismissed();
+                        }
+                      },
+                    ),
+                  ),
+                  Container(
+                    color: footerColor,
+                    height: mediaQueryData.padding.bottom,
```

Comment:
You can just use padding on the container above instead of a different widget. Semantically easier to read.

Reply:
This is needed because we need to draw in the bottom space on iOS and we can't use the other container because the button has cut outs